### PR TITLE
Exception handling

### PR DESCRIPTION
**Pull request by anmolbabu**

### CODE REVIEW

**Comment by anmolbabu** [comment]
This PR is about commits "Fix broad Exception handling" and "Channelise all etcd reads through common path" rebased on all currently open PRs, in order:
1. https://github.com/Tendrl/performance-monitoring/pull/197
2. https://github.com/Tendrl/performance-monitoring/pull/203
3. https://github.com/Tendrl/performance-monitoring/pull/205
This PR is not yet verified...

**Comment by r0h4n** [comment]
Resolve merge conflicts

File: tendrl/performance_monitoring/aggregator/cluster_summary.py [hunk 1]
```diff
@@ -34,7 +34,10 @@ def parse_host_count(self, cluster_id):
                         node_id
                     )
                 )
-            except EtcdKeyNotFound as ex:
+            except (
+                EtcdKeyNotFound,
+                TendrlPerformanceMonitoringException
```
Comment (r0h4n) [inline tendrl/performance_monitoring/aggregator/cluster_summary.py]: TendrlPerformanceMonitoringException is only a wrapper around Exception

Are you sure you have understood "Dont catch broad exceptions" ?

**Comment by anmolbabu** [comment]
Needs to be verified after rebase

**Comment by anmolbabu** [comment]
Verified on a buggy system...
Need to verify best case scenario...

Now TendrlPerformanceMonitoringException is used only in those places in code where we throwing our custom exceptions and not used as a wrapper over other exceptions anymore...

**Comment by anmolbabu** [comment]
Verified

**Pull request merged**